include/zephyr/drivers/w1.h

```diff
@@ -237,7 +237,7 @@ static inline int z_impl_w1_read_byte(const struct device *dev)
 {
 	const struct w1_driver_api *api = (const struct w1_driver_api *)dev->api;
 
-	return api->read_byte(dev);
+	return api->read_byte ? api->read_byte(dev) : -ENOSYS;
```

Comment:
So what's your suggestion? Drop this mechanism altogether? I think its='s generally a good idea to return something meaningful instead of crashing, but since the parent driver no longer uses the 1-Wire API this could be dropped.